UnitTests/CommonTestUtils/NoAssertContext.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System
+{
+    /// <summary>
+    ///  Use (within a using) to eat asserts.
+    /// </summary>
+    public sealed class NoAssertContext : IDisposable
```

Comment:
This is not self-explaining to me.

Why should `ParseCommitBody_should_return_null_if_no_EndOfBody` eat which asserts?

Reply:
We need to "eat" this assert:
https://github.com/gitextensions/gitextensions/blob/9564bb2ab333735141f767bdc73ee62501c11a05/GitCommands/RevisionReader.cs#L457
which otherwise could break tests or show an interactive dialog expecting a user input